Sources/SocksCore/Conversions.swift

```diff
@@ -20,9 +20,30 @@ extension Array {
 }
 
 extension String {
-    
     public func toBytes() -> [UInt8] {
         return Array(self.utf8)
     }
 }
 
+// Sendable and Receivable implementations
+
+extension String: Sendable {
+    public func serialize() -> [UInt8] {
+        return self.toBytes()
+    }
+}
+
+extension String: Receivable {
+    static func deserialize(reader: (maxBytes: Int) throws -> [UInt8]) throws -> String {
```

Comment:
I don't see where this is defined, but I have two things that might be nice additions.

1. Can we rename to `parse`. I think there's a lot of libraries (swift and otherwise) that have opted for `serialize/parse` over `serialize/deserialize`

2. Can `reader` be `@noescape`, it's already fulfilling the role, so this will allow us to use `self.` w/o having capture warnings etc.